packages/pretty-format/src/plugins/ImmutableList.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ * @flow
+ */
+
+'use strict';
+
+const printImmutable = require('./printImmutable');
+const IS_LIST_SENTINEL = '@@__IMMUTABLE_LIST__@@';
+
+const isList = (maybeList: Object) => {
+  return !!(maybeList && maybeList[IS_LIST_SENTINEL]);
+};
+
+const test = (object: Object) => object && isList(object);
+
+const print = (
+  val: Object, 
+  print: Function,
+  indent: Function,
+  opts: Object,
+  colors: Object
```

Comment:
the flow typing here could be a lot stronger. @cpenarrieta we just added flow types to pretty-format itself – mind sending a second, follow-up PR, where you use the types from the main module? You can extract it into a types.js file like we do in other packages.